cpp/src/arrow/ipc/writer.cc

```diff
@@ -828,12 +828,11 @@ Status WriteSparseTensor(const SparseTensor& sparse_tensor, io::OutputStream* ds
 Status GetSparseTensorMessage(const SparseTensor& sparse_tensor, MemoryPool* pool,
                               std::unique_ptr<Message>* out) {
   internal::IpcPayload payload;
-  RETURN_NOT_OK(internal::GetSparseTensorPayload(sparse_tensor, pool, &payload));
-
-  const std::shared_ptr<Buffer> metadata = payload.metadata;
-  const std::shared_ptr<Buffer> buffer = *payload.body_buffers.data();
+  std::shared_ptr<Buffer> buffer;
 
-  out->reset(new Message(metadata, buffer));
+  RETURN_NOT_OK(internal::GetSparseTensorPayload(sparse_tensor, pool, &payload));
+  RETURN_NOT_OK(arrow::ConcatenateBuffers(payload.body_buffers, pool, &buffer));
```

Comment:
I think this approach is not good because ConcatenateBuffers makes buffers' copy.

I'm trying to rewrite this without making buffer's copy.
Please wait for a moment.

Reply:
Thanks! I gave you access to my fork if you want to add the change here.